contracts/migrations/4_user_replica_set_migration.js

```diff
@@ -0,0 +1,80 @@
+const contractConfig = require('../contract-config.js')
+const Registry = artifacts.require('Registry')
+const UserReplicaSetManager = artifacts.require('UserReplicaSetManager')
+const AudiusAdminUpgradeabilityProxy2 = artifacts.require('AudiusAdminUpgradeabilityProxy2')
+const userFactoryKey = web3.utils.utf8ToHex('UserFactory')
+const userReplicaSetManagerKeyString = 'UserReplicaSetManager'
+const userReplicaSetManagerKey = web3.utils.utf8ToHex(userReplicaSetManagerKeyString)
+const abi = require('ethereumjs-abi')
+
+// Generate encoded arguments for proxy initialization
+const encodeCall = (name, args, values) => {
+  const methodId = abi.methodID(name, args).toString('hex')
+  const params = abi.rawEncode(args, values).toString('hex')
+  return '0x' + methodId + params
+}
+
+module.exports = (deployer, network, accounts) => {
+  deployer.then(async () => {
+    // TODO: Consider how this migration will be run against prod
+    //       Registry.deployed() may not be the cleanest way to do this
+    let registry = await Registry.deployed()
```

Comment:
Registry.deployed() may not be the cleanest way to do this - last time we discussed we agreed to resume the migration from the initial deployment. If that's the case this should work as is but its worth revisiting this topic. Alternative would be to consume an env var if present for Registry.at(address) or something diff

Reply:
i like how you modified the service-commands to specify migration number ranges to run, which makes this super simple. not really sure what `.deployed()` would do in a non-local env - ie if there are multiple deployed on ropsten which would it select? - how do you feel aobut just accepting registry address by environment as an additional config var

Reply:
i still think this is a risky way to perform the actual migration and would prefer to explicitly configure the registry address

Reply:
you planning on making this change? either way lets clean up this comment